### ISSUE TITLE
Form elements do not announce themselves properly

### ISSUE THREAD

**Issue by douglasbouttell-camunda** [bpmn-io/form-js#1042]
### Is your feature request related to a problem? Please describe.

When using a screenreader, form-js does not do the following:
* Announce the descriptions of any fields
* Announce the name of a group
* Announce if a field is required
* Announce if a field is invalid

Additionally, form-js will announce elements of the form which are purely visual. For example, it will announce the asterisk which denotes a required field.

### Describe the solution you'd like

Add in the appropriate HTML and ARIA labels.

### Describe alternatives you've considered

None

### Additional context

![image](https://github.com/bpmn-io/form-js/assets/155977861/e53295c9-eed7-4528-a555-8b6f79dbf189)
Announces `Creditor *, edit text`. `*`/"star" is a purely visual element. Does not announce that the field is required and invalid.

![image](https://github.com/bpmn-io/form-js/assets/155977861/19b18d13-018d-4e12-87b8-896b23337965)
Voiceover does not show this field as having a description when it does visually.

![image](https://github.com/bpmn-io/form-js/assets/155977861/3992f71f-19ac-47a2-b01a-3343b5c14682)
Does not announce the group name on entering.
